lighthouse-core/scripts/pptr-run-devtools.js

```diff
@@ -152,122 +176,146 @@ const startLighthouse = `
       }
     }
   }
+}
+
+async function runLighthouse() {
+  // @ts-expect-error global
+  const panel = UI.panels.lighthouse || UI.panels.audits;
+
+  /** @type {Promise<{lhr: LH.Result, artifacts: LH.Artifacts}>} */
+  const resultPromise = new Promise((resolve, reject) => {
+    const methodName = panel.__proto__.buildReportUI ?
+      'buildReportUI' : '_buildReportUI';
+    addSniffer(
+      panel.__proto__,
+      methodName,
+      // @ts-expect-error implicit any
+      (lhr, artifacts) => resolve({lhr, artifacts})
+    );
+
+    addSniffer(
+      panel.statusView.__proto__,
+      'renderBugReport',
+      // @ts-expect-error implicit any
+      (err) => reject(err)
+    );
+  });
 
+  const button = panel.contentElement.querySelector('button');
   button.click();
-})()
-`;
+
+  return resultPromise;
+}
+/* eslint-enable */
 
 /**
- * @param {string} url
+ * @param {puppeteer.Browser} browser
+ * @param {puppeteer.CDPSession} inspectorSession
+ * @param {LH.Config.Json} config
  */
-function isValidUrl(url) {
-  try {
-    new URL(url);
-    return true;
-  } catch {
-    return false;
+async function installCustomConfig(browser, inspectorSession, config) {
```

Comment:
I noticed that the custom config was not always being applied in time for Lighthouse to use it sometimes. This code has been refactored to (hopefully) prevent this race condition.